internal/vfs/vfs.go

```diff
@@ -266,6 +266,7 @@ func (q *QuotaCheckResult) GetRemainingFiles() int {
 type S3FsConfig struct {
 	sdk.BaseS3FsConfig
 	AccessSecret *kms.Secret `json:"access_secret,omitempty"`
+	SessionToken string      `json:"session_token,omitempty"`
```

Comment:
The session token is added as an HTTP header or as a query string parameter, so differently from the `AccessSecrer` we can store it as a string, but it should be added to `sdk.BaseS3FsConfig` (and probably also the terraform provider should be updated).